assets/js/components/legacy-notifications/dashboard-modules-alerts.js

```diff
@@ -20,91 +20,86 @@
  * External dependencies
  */
 import { each } from 'lodash';
+import { useMount } from 'react-use';
 
 /**
  * WordPress dependencies
  */
-import { Component, Fragment } from '@wordpress/element';
+import { Fragment, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import { getModulesData } from '../../util';
+import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import NotificationAlertSVG from '../../../svg/notification-alert.svg';
 import Notification from './notification';
 import { modulesNotificationsToRequest, getModulesNotifications } from './util';
-import NotificationAlertSVG from '../../../svg/notification-alert.svg';
+const { useSelect } = Data;
 
-class DashboardModulesAlerts extends Component {
-	constructor( props ) {
-		super( props );
+function DashboardModulesAlerts() {
+	const [ alerts, setAlerts ] = useState( [] );
 
-		this.state = {
-			data: false,
-		};
-	}
+	const modules = useSelect( ( select ) =>
+		select( CORE_MODULES ).getModules()
+	);
 
-	componentDidMount() {
-		const modules = modulesNotificationsToRequest();
-
-		if ( modules ) {
-			getModulesNotifications( modules ).then( ( response ) => {
-				this.setState( {
-					data: response.results,
-				} );
-			} );
-		}
-	}
+	useMount( async () => {
+		const modulesWithNotifications = modulesNotificationsToRequest();
 
-	render() {
-		const { data } = this.state;
+		if ( modulesWithNotifications ) {
+			const response = await getModulesNotifications(
+				modulesWithNotifications
+			);
```

Comment:
I realize this is how it was before, but `getModulesNotifications` actually takes no arguments and instead calls `modulesNotificationsToRequest` internally 😄 
Maybe best to keep that way since the return value from that function is an odd structure.